crates/indexer/src/models/token_models/v2_token_ownerships.rs

```diff
@@ -164,6 +164,10 @@ impl TokenOwnershipV2 {
 
         // check if token was transferred
         if let Some((event_index, transfer_event)) = &metadata.transfer_event {
+            // If it's a self transfer then skip
+            if transfer_event.get_to_address() == transfer_event.get_from_address() {
```

Comment:
Hmm it seems a bit hacky that this needs a special case. Shouldn't this still work if the to address is still this account? I.e. It seems like the code is asymmetric and deleting ownership for from but don't really update it for to?

Reply:
It's the ordering right? If I move this code before the previous section, it'll be fine. However, IMO explicitly calling out this edge case is the safer route. I mean there really aren't that many cases in a transfer: transfer to self, transfer to someone else.